ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/CaseMapper.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.cmc.ccd.mapper;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCase;
+import uk.gov.hmcts.cmc.ccd.domain.CCDDocument;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+
+import java.net.URI;
+import java.util.Arrays;
+
+import static uk.gov.hmcts.cmc.ccd.domain.CCDYesNoOption.NO;
+import static uk.gov.hmcts.cmc.ccd.domain.CCDYesNoOption.YES;
+
+@Component
+public class CaseMapper {
+
+    private final ClaimMapper claimMapper;
+
+    public CaseMapper(ClaimMapper claimMapper) {
+        this.claimMapper = claimMapper;
+    }
+
+    public CCDCase to(Claim claim) {
+        final CCDCase.CCDCaseBuilder builder = CCDCase.builder();
+
+        if (claim.getLetterHolderId() != null) {
+            builder.letterHolderId(claim.getLetterHolderId());
+        }
+
+        if (claim.getDefendantId() != null) {
+            builder.defendantId(claim.getDefendantId());
+        }
+
+        claim.getSealedClaimDocument().ifPresent(document -> builder
+            .sealedClaimDocument(CCDDocument.builder()
+                .documentUrl(document.toString())
+                .build())
+        );
+
+        claimMapper.to(claim.getClaimData(), builder);
+
+        return builder
```

Comment:
Since its builder and we are already doing mapping using builder above. Do all the mapping  and then last return to be just "return builder.build". Code will be more readable.

Reply:
very old code, currently it does